SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -242,7 +242,8 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 	// If arguments can not be found we will use the regular flow of loading from the input URL.
 
 	FString SpatialWorkerType = GetGameInstance()->GetSpatialWorkerType().ToString();
-	if (!GameInstance->GetFirstConnectionToSpatialOSAttempted() && !GameInstance->GetPreventAutoConnectWithLocator())
+
+	if (!GameInstance->GetFirstConnectionToSpatialOSAttempted() && !GetPreventAutoConnectWithCommandLineArgs())
```

Comment:
So if a client sets `bPreventAutoConnectWithCommandLineArgs` to `true` it won't go through this first if statement. 
But, this is the only place where `GameInstance->SetFirstConnectionToSpatialOSAttempted();` is called.
This makes me think `GetFirstConnectionToSpatialOSAttempted()` is now a bad function since it's value isn't always correct.

Reply:
I think having GetFirstConnectionToSpatialOSAttempted is still valid. I think I just need to re-arrange the logic to ensure SetFirstConnectionToSpatialOSAttempted is always called on first connection.